typeset/derive/config.satyg

```diff
@@ -1,14 +1,104 @@
 @import: line
 
-type label-position =
-    | LabelLeft
-    | LabelRight
+type label-pos-t =
+| LabelLeft
+| LabelRight
 
-type derive-config = (|
-    default-label-position : label-position;
-    default-line : derivation-line;
+type derive-config-t = (|
+    default-label-position : label-pos-t;
+    default-line : DeriveLine.u;
     bar-top-bottom-margin : length;
     label-left-right-margin : length;
     bar-thickness : length;
     content-left-right-padding : length;
 |)
+
+module DeriveConfig : sig
+
+    val default : derive-config-t
+
+    val with-default-label-position : label-pos-t -> derive-config-t -> derive-config-t
+    val with-default-line : DeriveLine.u -> derive-config-t -> derive-config-t
+    val with-bar-top-bottom-margin : length -> derive-config-t -> derive-config-t
+    val with-label-left-right-margin : length -> derive-config-t -> derive-config-t
+    val with-bar-thickness : length -> derive-config-t -> derive-config-t
+    val with-content-left-right-padding : length -> derive-config-t -> derive-config-t
+
+end = struct
+
+    type t = (|
+        default-label-position : label-pos-t;
+        default-line : DeriveLine.u;
+        bar-top-bottom-margin : length;
+        label-left-right-margin : length;
+        bar-thickness : length;
+        content-left-right-padding : length;
+    |)
+
+
+    let default = (|
```

Comment:
Perhaps you can use record update syntax https://github.com/gfngfn/SATySFi/commit/850ba23a4da0271033181496498d8473be2b7445